File: src/ProjectR/content/scripts/spells/OmaeWaMoShindeiru.lua
function GetTargetType()
    return Single
end

function GetName()
  return "Omae Wa Mo Shindeiru"
end

function GetDescription()
  return "You are already dead."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 1
end

function GetDelay()
  return 0
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return { DRK }
end

function SpellEffect()
  local bonusDamage = 1
  if d:GetCurrentHP() / dHP < .5 then
    bonusDamage = 3
  end

  local damage = (5 * aAD * (aDRK/100) - dDEF) * (100/dDRK) * bonusDamage
  d:ApplyDebuff(DTH, 75)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/OmaeWaMoShindeiru.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class OmaeWaMoShindeiru : SpellScriptBase
    {
        public override string Name { get { return "Omae Wa Mo Shindeiru"; } }
        public override string Description { get { return "You are already dead."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.DRK, }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 100; } }
        public override double Delay { get { return 0; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var bonusDamage = target.CurrentHP / tHP < .5 ? 3 : 1;

            var damage = (5 * cAD * (cDRK/100) - tDEF) * (100/tDRK) * bonusDamage;
            TryToApplyDebuff(DebuffResistance.DTH, 75);
            DealDamage(damage);
        }
    }
}